### PR TITLE
Improve configurations

### PR DESCRIPTION
**Pull request by LoicGrobol**
This is an attempt at simplifying how the models and configuration work, making some things more explicit and adding some new options.

**Note** There is a slight breaking change here: model subfiles have to be renamed for old models to work so the currently provided models should be repackaged. ~~It should probably be done before merging this but I can do it. Also according to [GitHub doc](https://docs.github.com/en/free-pro-team@latest/github/managing-large-files/distributing-large-binaries), we can upload files up to 2 GiB in releases, so at least the non-BERT models could go there, and the others either on Sharedoc or Zenodo if they are too large.~~ 

The repackaged non-BERT models have been uploaded to [a pre-release](https://github.com/bencrabbe/npdependency/releases/tag/v0.1.0dev0) that serves as on-GitHub hosting. The BERT models are slightly over 2GiB/file so they couldn't be put there. As a temporary solution, I have put them in [a public directory](https://sharedocs.huma-num.fr/wl/?id=Kh5C5FFdhMInTWgUQwLnWof9fXDlM4al) on HumaNum's Sharedoc. Models for a stable release should probably go to Zenodo, where they can get a DOI and are more easily cited.

### CLI

- **Breaking** An `--overwrite` argument to force retraining the model from scratch if one is already present in the target directory
- A `--fasttext` argument to `graph_parser` that can take either the path to a pretrained FastText model or the path to a raw text file to generate a model from.
- A `--device` argument, to override the device specified in a model configuration
- More explicit logging of what is done with FastText models at train time, and in particular whether the model is loaded or trained and the path to the source files

### Models

- Configs now accept a `cased` parameter to indicate whether a BERT lexer should be cased. If this parameter is absent, this choice defaults to `cased`, unless the BERT model has `uncased` in its name (the previous behaviour).

### API

- A new classmethod `graph_parser.BiaffineParser.from_config` that wraps the whole process of creating a `BiaffineParser` from a config directory with a config file and all the vocabulary and auxiliary models (but possibly lacking the weights).
- Two new functions `deptree.gen_tags` and `deptree.gen_labels` to generate tags and labels vocabularies without having to create a full `DependencyDataset`
- A new class method `lexers.FastTextTorch.train_model_from_raw`, a simple wrapper for `fasttext.train_unsupervised`

### CLI

- **Breaking** When `--overwrite` is not provided and a trained model already exists in the destination directory, training continues from its state instead of starting from scratch
- Model trained on a device can now be loaded on another device if their `device` config parameter is changed or when asked using `--device`

### Models

- **Breaking** Model subfiles (apart from the config file) now use static names (like `charcodes.lst`), to simplify model saving and loading, the assumption being that since these files are tied to a specific model directory anyway, they don't need custom names. Old models can still be used by simply renaming their subfiles.
- Model configurations can now omit the `device` parameter, which then defaults to `cpu`

### API

- `lexers.FastTextTorch.train_model` has been renamed to `lexers.FastTextTorch.train_model_from_trees` and now throws an error if the destination files already exists instead of silently loading.

### Misc

- Model initialization and loading now both use  `graph_parser.BiaffineParser.from_config` to avoid error-prone code duplication.
- Config files are now loaded with the pyyaml `SafeLoader`, since we do not actually need more than that for now.

### CODE REVIEW

**Comment by LoicGrobol** [comment]
I have been dogfooding this for some time and I think it's ready to merge now that the repacked models are online.

**Comment by bencrabbe** [comment]
Wonderful Loic, this looks great ! I merge. One question remains : why did you choose to put the fasttext flag as a flag rather than as a field in the config file ?

**Comment by Evpok** [comment]
@bencrabbe A great question, which I have thought about a lot :-) in the end I decided to make it a flag because it is only used in train mode and ignored at test time, so to me, leaving it in the configuration would have been confusing. Of course, this is already the case of the pretraining hyperparameters so…
In any case, it can be changed if you want. I could also be nice to have a chat about training configs and model files when we next get some time without being rushed by sanitary upheavals and conf deadlines…

**Comment by LoicGrobol** [comment]
@bencrabbe A great question, which I have thought about a lot :-) in the end I decided to make it a flag because it is only used in train mode and ignored at test time, so to me, leaving it in the configuration would have been confusing. Of course, this is already the case of the pretraining hyperparameters so…
In any case, it can be changed if you want. I could also be nice to have a chat about training configs and model files when we next get some time without being rushed by sanitary upheavals and conf deadlines…
